vault/core.go

```diff
@@ -131,6 +131,19 @@ type unlockInformation struct {
 	Nonce string
 }
 
+type counters struct {
+	// activePath is set at startup to the path we primed the requests counter from,
+	// or empty string if there wasn't a relevant path - either because this is the first
+	// time Vault starts with the feature enabled, or because Vault hadn't written
+	// out the request counter this month yet.
+	// Whenever we write out the counters, we update activePath if it's no longer
+	// accurate.  This coincides with a reset of the counters.
+	// There's no lock because the only reader/writer of activePath is the goroutine
+	// doing background syncs.
+	activePath string
+	requests   uint64
```

Comment:
Let's make this a `*uint64` Also for this to work on 32-bit operating systems we need to put it as the first item in the struct to properly align it. See https://golang.org/pkg/sync/atomic/#pkg-note-BUG for more info

Reply:
Ah, good catch re alignment, thanks.

I don't mind making it a pointer, but could you explain why for future reference?

Reply:
Mostly just based off the usage pattern, we only want to use this with atomic calls. So by making it a pointer it removes the requirement of explicitly referencing it on each call. Also it makes it a little more clear that you shouldn't manipulate the value outside of atomic calls

Reply:
Makes sense, ty.